From https://qa-proxy.ceph.com/teuthology/dgalloway-2026-06-01_17:00:02-rados-issue-75389-yaml-and-jinja2-deps-on-centos-distro-distro-default-trial/252250/teuthology.log:

```
2026-06-01T17:12:32.093 INFO:tasks.workunit.client.0.trial169.stdout:cephadm/build.py: Running cmd: /tmp/tmp62o057lm.cephadm.build/deps/_venv_/bin/python3 -m pip list --format=json --path /tmp/tmp62o057lm.cephadm.build/deps
2026-06-01T17:12:32.206 INFO:tasks.workunit.client.0.trial169.stdout:cephadm/build.py: Copying contents
2026-06-01T17:12:32.213 INFO:tasks.workunit.client.0.trial169.stdout:cephadm/build.py: Byte-compiling py to pyc
2026-06-01T17:12:32.304 INFO:tasks.workunit.client.0.trial169.stdout:cephadm/build.py: Constructing the zipapp file
2026-06-01T17:12:32.365 INFO:tasks.workunit.client.0.trial169.stdout:cephadm/build.py: Zipapp created with compression
2026-06-01T17:12:32.383 INFO:tasks.workunit.client.0.trial169.stderr:+ sudo tmp.test_repos.sh.B5WYzW/tmp.cephadm.BzQncA -v add-repo --release quincy
2026-06-01T17:12:32.480 INFO:tasks.workunit.client.0.trial169.stderr:--------------------------------------------------------------------------------
2026-06-01T17:12:32.480 INFO:tasks.workunit.client.0.trial169.stderr:cephadm ['-v', 'add-repo', '--release', 'quincy']
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:unable to fetch repo metadata: <HTTPError 404: 'Not Found'>
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:Traceback (most recent call last):
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:  File "/tmp/tmp62o057lm.cephadm.build/app/cephadmlib/packagers.py", line 428, in validate
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:  File "/usr/lib64/python3.12/urllib/request.py", line 215, in urlopen
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:    return opener.open(url, data, timeout)
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:  File "/usr/lib64/python3.12/urllib/request.py", line 521, in open
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:    response = meth(req, response)
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:               ^^^^^^^^^^^^^^^^^^^
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:  File "/usr/lib64/python3.12/urllib/request.py", line 630, in http_response
2026-06-01T17:12:32.758 INFO:tasks.workunit.client.0.trial169.stderr:    response = self.parent.error(
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:               ^^^^^^^^^^^^^^^^^^
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "/usr/lib64/python3.12/urllib/request.py", line 559, in error
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:    return self._call_chain(*args)
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:           ^^^^^^^^^^^^^^^^^^^^^^^
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "/usr/lib64/python3.12/urllib/request.py", line 492, in _call_chain
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:    result = func(*args)
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:             ^^^^^^^^^^^
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "/usr/lib64/python3.12/urllib/request.py", line 639, in http_error_default
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:    raise HTTPError(req.full_url, code, msg, hdrs, fp)
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:urllib.error.HTTPError: HTTP Error 404: Not Found
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:During handling of the above exception, another exception occurred:
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:Traceback (most recent call last):
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "<frozen runpy>", line 198, in _run_module_as_main
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "<frozen runpy>", line 88, in _run_code
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "/tmp/tmp62o057lm.cephadm.build/app/__main__.py", line 5747, in <module>
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "/tmp/tmp62o057lm.cephadm.build/app/__main__.py", line 5735, in main
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "/tmp/tmp62o057lm.cephadm.build/app/__main__.py", line 4527, in command_add_repo
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:  File "/tmp/tmp62o057lm.cephadm.build/app/cephadmlib/packagers.py", line 431, in validate
2026-06-01T17:12:32.759 INFO:tasks.workunit.client.0.trial169.stderr:cephadmlib.exceptions.Error: failed to fetch repository metadata. please check the provided parameters are correct and try again
```

Seems related to this PR.

--- a/ceph.spec.in
+++ b/ceph.spec.in
@@ -558,13 +558,39 @@ Requires:       which
 %if 0%{?weak_deps}
 Recommends:     podman >= 2.0.2
 %endif
+# Cephadm zipapp: CMake sets CEPHADM_BUNDLED_DEPENDENCIES to pip, rpm, or none
+# (see build / cmake block below):
+#  - with cephadm_bundling, with cephadm_pip_deps: pip at build, no stanzas here
+#  - with cephadm_bundling, without cephadm_pip_deps: build-time RPM deps only
+#  - without cephadm_bundling: unbundled cephadm, runtime Requires for yaml/jinja2
+# CentOS Storage SIG #75389 needs the unbundled branch (--without cephadm_bundling)
+# after dropping downstream patch 0036; pip-mode SIG builds are a separate SIG fix.
 %if 0%{with cephadm_bundling}
 %if 0%{without cephadm_pip_deps}
-BuildRequires: python3-jinja2 >= 2.10
-%endif
+# Zipapp built from system RPMs at build time; bundled zipapp is self-contained at runtime.
+# SUSE and RHEL/Fedora use different PyPI RPM names (Jinja2/PyYAML vs jinja2/pyyaml).
+%if 0%{?suse_version}
+BuildRequires: python%{python3_pkgversion}-Jinja2 >= 2.10
+BuildRequires: python%{python3_pkgversion}-PyYAML
 %else
-Requires: python3-jinja2 >= 2.10
+BuildRequires: python%{python3_pkgversion}-jinja2 >= 2.10
+BuildRequires: python%{python3_pkgversion}-pyyaml
 %endif
+%dnl suse/else: rpm bundle, distinct RPM names only (not duplicate stanzas)
+%endif
+%dnl end without cephadm_pip_deps (CEPHADM_BUNDLED_DEPENDENCIES=rpm)
+%else
+# Unbundled cephadm: host must provide yaml and jinja2 at runtime.
+%if 0%{?suse_version}
+Requires: python%{python3_pkgversion}-Jinja2 >= 2.10
+Requires: python%{python3_pkgversion}-PyYAML
+%else
+Requires: python%{python3_pkgversion}-jinja2 >= 2.10
+Requires: python%{python3_pkgversion}-pyyaml
+%endif
+%dnl suse/else: unbundled cephadm (CEPHADM_BUNDLED_DEPENDENCIES=none)
+%endif
+%dnl end with cephadm_bundling
 %description -n cephadm
 Utility to bootstrap a Ceph cluster and manage Ceph daemons deployed
 with systemd and podman.

